File: android/LocationTracker/src/main/java/ca/mcmaster/capstone/monitoralgorithm/AutomatonState.java
package ca.mcmaster.capstone.monitoralgorithm;

import lombok.AllArgsConstructor;
import lombok.NonNull;
import lombok.Value;

/* Class to represent an automaton state.*/
@Value @AllArgsConstructor
public class AutomatonState {
    @NonNull String stateName;
    @NonNull Automaton.Evaluation stateType;

    public AutomatonState(@NonNull final AutomatonState state) {
        this.stateName = state.stateName;
        this.stateType = state.stateType;
    }
}

File: ios/capstone/capstone/AutomatonState.swift
//
//  AutomatonState.swift
//  capstone
//
//  Created by Darren Kitamura on 2015-01-15.
//  Copyright (c) 2015 DKitamura. All rights reserved.
//

import Foundation


class AutomatonState {
    var stateName: String
    var stateType: Automaton.Evaluation
    
    init(state: AutomatonState) {
        self.stateName = state.stateName
        self.stateType = state.stateType
    }
    
    func toString() -> String {
        return "AutomatonState{stateName='\(stateName)', stateType=\(stateType)}"
    }
    //TODO Override Equals...
    
    
    //TODO Hashcode
    func hashCode() -> Int {
        var result: Int
        
        return 0
    }
    
}